hnn_core/cell.py

```diff
@@ -184,27 +184,22 @@ def insert_dipole(self, yscale):
 
         # dends must have already been created!!
         # it's easier to use wholetree here, this includes soma
-        seclist = h.SectionList()
-        seclist.wholetree(sec=self.soma)
-        # create a python section list list_all
-        list_all = [sec for sec in seclist]
-        for sect in list_all:
+        sec_list = h.SectionList()
+        sec_list.wholetree(sec=self.soma)
+        sec_list = [sec for sec in sec_list]
```

Comment:
Can this just be a oneliner?
```py
sec_list = [sec for sec in h.SectionList().wholetree(sec=self.soma)]
```
The immediate conversion from hoc to python objects seems weird.

Reply:
unfortunately we can't do that ... for chaining to work `wholetree(...)` must return an iterable. Instead it modifies the object in place and returns a float :(

Reply:
we could monkey-patch some of the objects and maintain them in a file called `fixes.py` ([see MNE for inspiration](https://github.com/mne-tools/mne-python/blob/master/mne/fixes.py)). I think there were some improvements to `Exp2Syn` as well. Once we have a sufficiently large number of improvements, we can submit them to the Neuron team.

Reply:
Definitely a good idea. Do you have a few other improvements in mind that would simplify our code? It'd be good to open a PR to make a list of QOL improvements, but it's probably only worth it if there's more than syntax nitpicks.